crates/oracle/src/subgraph.rs

```diff
@@ -46,18 +53,47 @@ impl From<&Config> for SubgraphQuery {
 
 #[async_trait]
 impl SubgraphApi for SubgraphQuery {
-    type State = Vec<subgraph_state::SubgraphStateNetworks>;
-
-    async fn get_subgraph_state(&self) -> anyhow::Result<Self::State> {
-        let response = query(self.url.clone())
-            .await
-            .context("querying Epoch Subgraph")?;
-        let ResponseData { networks } = response;
-        Ok(networks)
+    type State = subgraph_state::SubgraphStateGlobalState;
+
+    async fn get_subgraph_state(&self) -> anyhow::Result<Option<Self::State>> {
+        Ok(query(self.url.clone()).await?)
+    }
+}
```

Comment:
This is fine for now, as it's not within the scope of the PR, but I think `SubgraphApi` should get discarded and become a simple function instead.

Reply:
The only reason this trait exist is to allow tests for the `SubgraphState` struct. 
Maybe we can think of some way to make it so without this trait.